src/main/java/de/tum/cit/aet/artemis/assessment/service/ResultService.java

```diff
@@ -522,23 +522,26 @@ private void handleFeedbackPersistence(Feedback feedback, Result result, Map<Lon
         // Temporarily detach feedback from the parent result to avoid Hibernate issues
         feedback.setResult(null);
 
-        // Clear the long feedback if it exists in the map
+        // Connect old long feedback text to the feedback before saving, otherwise it would be deleted
         if (feedback.getId() != null && feedback.getHasLongFeedbackText()) {
-            LongFeedbackText longFeedback = longFeedbackTextMap.get(feedback.getId());
-            if (longFeedback != null) {
-                feedback.clearLongFeedback();
+
+            // If the long feedback is not empty, it means that changes have been made on the client, so we do not want
+            // to override the new long feedback with its previous version
+            if (feedback.getLongFeedback().isPresent()) {
+                // Delete the old long feedback so we don't get a duplicate key error
+                longFeedbackTextRepository.deleteByFeedbackIds(List.of(feedback.getId()));
```

Comment:
```suggestion
                longFeedbackTextRepository.deleteByFeedbackId(feedback.getId());
```
It should probably be enough to add to the repository
```java
@Modifying
@Transactional
void deleteByFeedbackId(Long feedbackId);
```
without requiring a custom query for this.

Reply:
Well I just used the one that was there already. I added it though, might be convenient to have anyway.